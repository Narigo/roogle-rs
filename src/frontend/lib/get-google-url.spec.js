import getGoogleUrl from "./get-google-url";

describe("get-google-url", () => {
  it("throws an error on no argument", () => {
    expect(() => getGoogleUrl()).toThrow();
  });

  it("throws an error when there is an empty string as argument", () => {
    expect(() => getGoogleUrl("")).toThrow();
  });

  it("returns the proper URL", () => {
    expect(getGoogleUrl("This is a test of the basic functionality")).toMatchSnapshot();
  });

  it("returns the proper URL when Umlaute are used", () => {
    expect(getGoogleUrl("Das ist ein SÃ¤tzchen mit Umlauten dass das mal getestet wird")).toMatchSnapshot();
  });
});
