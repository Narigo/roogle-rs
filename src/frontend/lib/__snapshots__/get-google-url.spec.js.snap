// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`get-google-url returns the proper URL 1`] = `"https://www.google.de/search?q=This%20is%20a%20test%20of%20the%20basic%20functionality"`;

exports[`get-google-url returns the proper URL when Umlaute are used 1`] = `"https://www.google.de/search?q=Das%20ist%20ein%20S%C3%A4tzchen%20mit%20Umlauten%20dass%20das%20mal%20getestet%20wird"`;
